longevity_tombstone_gc_test.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import time
+from functools import partial
+
+from longevity_twcs_test import TWCSLongevityTest
+from sdcm.utils.common import ParallelObject
+from sdcm.utils.sstable_utils import SstableUtils
+
+
+class TombstoneGcLongevityTest(TWCSLongevityTest):
+    keyspace = 'scylla_bench'
+    table = 'test'
+    ks_cf = f'{keyspace}.{table}'
+    propagation_delay = 60 * 4  # Setting a value shorter than Scylla's default (1 hour) - to 4 minutes.
+    repair_date = None
+    db_node = None
+
+    def _run_repair_and_major_compaction(self, wait_propagation_delay: bool = False):
+        self.log.info('Run a flush for %s on nodes', self.keyspace)
+        triggers = [partial(node.run_nodetool, sub_cmd=f"flush -- {self.keyspace}", ) for node in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+
+        self.log.info('Run a repair for %s on nodes', self.ks_cf)
+        triggers = [partial(node.run_nodetool, sub_cmd="repair", args=f"-pr {self.keyspace} {self.table}", ) for node
+                    in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+
+        if wait_propagation_delay:
+            time.sleep(self.propagation_delay)
+        self.repair_date = datetime.datetime.now()
+        self.log.info('Run a major compaction for user-table on node')
+        self.db_node.run_nodetool("compact", args=f"{self.keyspace} {self.table}")
+        self.wait_no_compactions_running()
+
+    def test_switch_tombstone_gc_modes(self):
+        """
+        test steps:
+        -----------
+        Based on TWCS TTL 48h longevity configuration with a much shorter TTL (few minutes) and same for gc-grace-seconds.
+        Start with tombstone-gc-mode disabled
+        wait a duration of TTL.
+        count number of tombstones.
+        run a repair + a major compaction.
+        count again to see not fewer tombstones exit.
+        change gc-grace-seconds back to default of 10 days.
+        change gc-mode to repair
+        run a repair + wait propagation delay + run a major compaction.
+        verify no tombstones exist in post-repair-created sstables.
+        change gc-mode to immediate.
+        wait for load to end + TTL period.
+        Run a major compaction.
+        Verify no tombstones.
+        """
+        # pylint: disable=too-many-locals
+
+        self.create_tables_for_scylla_bench()
```

Comment:
The table will be created with STCS for a tested OSS version or ICS if tested with an enterprise version.
Not sure what's best or how interesting is to test other compaction strategies.

Reply:
i think that it will be good for now, to test STCS on OSS and ICS on enterprise

Reply:
a correction, i forgot it already configures TWCS hardcoded in it.
It can be overridden by the `post_prepare_cql_cmds` value.
I tested both TWCS and LCS, not sure, what's the best option, need a feedback from @roydahan as well.